ProjFS.Mac/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```diff
@@ -204,29 +256,169 @@ static bool TryFetchAndLogKextHealthData(io_connect_t connection)
+    switch (messageLogType)
+    {
+        case OS_LOG_TYPE_ERROR:
+        {
+            JsonWriter messageWriter;
+            messageWriter.Add(MessageKey, string(message, messageLength));
+            WriteJsonToMessageListener(ErrorMessageEventName, messageWriter);
+            break;
+        }
+
+        default:
+            break;
+    }
+}
+
+static void LogKextHealthData(const PrjFSVnodeCacheHealth& healthData)
+{
+    os_log(
+        s_kextLogger,
+        "PrjFS Vnode Cache Health: CacheCapacity=%u, CacheEntries=%u, InvalidationCount=%llu, CacheLookups=%llu, LookupCollisions=%llu, FindRootHits=%llu, FindRootMisses=%llu, RefreshRoot=%llu, InvalidateRoot=%llu",
+        healthData.cacheCapacity,
+        healthData.cacheEntries,
+        healthData.invalidateEntireCacheCount,
+        healthData.totalCacheLookups,
+        healthData.totalLookupCollisions,
+        healthData.totalFindRootForVnodeHits,
+        healthData.totalFindRootForVnodeMisses,
+        healthData.totalRefreshRootForVnode,
+        healthData.totalInvalidateVnodeRoot);
+    
+    JsonWriter healthDataWriter;
+    healthDataWriter.Add(MessageKey, "Vnode cache health");
+    healthDataWriter.Add("CacheCapacity", healthData.cacheCapacity);
+    healthDataWriter.Add("CacheEntries", healthData.cacheEntries);
+    healthDataWriter.Add("InvalidationCount", healthData.invalidateEntireCacheCount);
+    healthDataWriter.Add("CacheLookups", healthData.totalCacheLookups);
+    healthDataWriter.Add("LookupCollisions", healthData.totalLookupCollisions);
+    healthDataWriter.Add("FindRootHits", healthData.totalFindRootForVnodeHits);
+    healthDataWriter.Add("FindRootMisses", healthData.totalFindRootForVnodeMisses);
+    healthDataWriter.Add("RefreshRoot", healthData.totalRefreshRootForVnode);
+    healthDataWriter.Add("InvalidateRoot", healthData.totalInvalidateVnodeRoot);
+    WriteJsonToMessageListener(HealthMessageEventName, healthDataWriter);
+}
+
+static void CreatePipeToMessageListener()
+{
+    lock_guard<mutex> lock(s_messageListenerMutex);
+    
+    if (INVALID_SOCKET_FD != s_messageListenerSocket)
+    {
+        // Already connected
+        return;
+    }
+
+    int error = 0;
+
+    s_messageListenerSocket = socket(PF_UNIX, SOCK_STREAM, 0);
+    if (s_messageListenerSocket < 0)
+    {
+        error = errno;
+        os_log(s_daemonLogger, "CreatePipeToMessageListener: Failed to create socket, error: %d", error);
+        s_messageListenerSocket = INVALID_SOCKET_FD;
+        return;
+    }
+    
+    struct sockaddr_un socket_address;
+    memset(&socket_address, 0, sizeof(struct sockaddr_un));
+    
+    socket_address.sun_family = AF_UNIX;
+    size_t pathLength = MessageListenerSocketPath.length();
+    if (pathLength + 1 >= sizeof(socket_address.sun_path))
+    {
+        os_log(s_daemonLogger, "CreatePipeToMessageListener: Failed to copy socket path, insufficient buffer. Buffer size: %lu", sizeof(socket_address.sun_path));
```

Comment:
This may be a nit, but some places in this commit you have %lu and some places have %llu.  Perhaps do a pass to double check these.  Thanks!

Reply:
Thanks! I'll switch the `size_t` format strings to use `%zu`